### PR TITLE
add Pipfile and Pipfile.lock

### LINKED ISSUE #194: Use pipenv instead of virtualenv
[Pipenv](https://pypi.org/project/pipenv/) has a few nice features over virtualenv like separating dev and prod Python dependencies. Pipenv can still generate a `requirements.txt` file if we want and Heroku supports it by default. It might be nice to transition to that tool and update the developer docs to use it instead of virtualenv.

### PR DESCRIPTION
**Pull request by cs-miller**
remove requirements.txt in favor of Pipfile/Pipfile.lock (https://pypi.org/project/pipenv/)
fixes #194

### CODE REVIEW

File: Pipfile
Comment (erwa) [inline Pipfile:18]: Is this Pipfile auto-generated? Can these packages be alphabetized?

**Review by erwa — COMMENTED** [review]
This looks good to me overall. Just a heads up, you may need to change the .circleci/config.yml (or the .github/workflows/pre-commit.yaml file after #233) since they both currently do pip install -r requirements.txt.

File: Pipfile
Comment (cs-miller) [reply Pipfile:18]: Yeah, it's autogenerated. For whatever reason, running pipenv install <x> does not alphabetize the Pipfile.
I will look into a solution for this.

Comment (cs-miller) [reply Pipfile:18]: I'm planning to update the dockerfile and ci before landing this

**Comment by cs-miller** [comment]
At the very least, I'm gonna wait on #233 so that I can update those config files

File: Pipfile
Comment (rgossiaux) [reply Pipfile:18]: +1 on updating the Docker config :)

**Comment by rgossiaux** [comment]
just curious, have you looked into poetry at all? Looks like another popular tool for package management. i don't really have much experience with either one personally

**Comment by cs-miller** [comment]
Yeah, I’m gonna take a look at it today

**Comment by akirabaruah** [comment]
> just curious, have you looked into poetry at all? Looks like another popular tool for package management. i don't really have much experience with either one personally

to add on, @cs-miller and I chatted and had some of these thoughts last night:
1. Heroku seems to support `requirements.txt` (`virtualenv`) or `Pipfile` (`pipenv`) out of the box (no config required). Likely not the case for poetry, but also not a huge deal to configure it accordingly.
2. poetry follows the `pyproject.toml` standard format, which is also used by `black` for config.

File: dev-guide.md
Comment (akirabaruah) [inline dev-guide.md:71]: We should update the docs below about calling `./manage.py ...`

(you already had plans for this, but commenting just for the record)

**Comment by akirabaruah** [comment]
> At the very least, I'm gonna wait on #233 so that I can update those config files

FYI this has landed :+1:

**Comment by cs-miller** [comment]
poetry seems to... not work? with fresh everything on WSL, so I'm not inclined to look into it very deeply.

**Comment by cs-miller** [comment]
I think I should probably update the dev guide a bit more, but as it stands everything "works"

**Comment by cs-miller** [comment]
i will say that its not super clear to me that this is a large improvement over the status quo, so i'm fine not merging this at all

**Comment by akirabaruah** [comment]
The separation of direct deps from transitive deps seems like a win IMO. With `requirements.txt`, didn't we find some dangling deps that weren't cleaned up when we removed some other main deps that brought them in indirectly?

Plus, you already put in the effort to get this working, so I'd advocate for merging this PR 😄

**Comment by cs-miller** [comment]
problem is it breaks people's existing workflow (must use pipenv). anyway, ill merge it if someone approves :)

**Review by akirabaruah — APPROVED** [review]
LGTM, but did we want to update the docs about using `./manage.py` commands with `pipenv`? Or at least remove references to `requirements.txt`?

**Comment by akirabaruah** [comment]
> problem is it breaks people's existing workflow (must use pipenv). anyway, ill merge it if someone approves :)

yeah, I think as long as we update the dev guide docs accordingly, it should be okay?

**Comment by rgossiaux** [comment]
> poetry seems to... not work? with fresh everything on WSL, so I'm not inclined to look into it very deeply.

Yeah seems fine, no need to look into it further. The Heroku support thing alone seems like a pretty compelling reason to use pipenv.

File: dev-guide.md
Comment (cs-miller) [reply dev-guide.md:78]: is this actually different from running whatever command is already in the docs (provided one is inside the env/shell)?

File: dev-guide.md
Comment (akirabaruah) [inline dev-guide.md:15]: Should we include a minimum version? Or at least say Python 3?

File: dev-guide.md
Comment (akirabaruah) [inline dev-guide.md:142]: Hmm, actually, maybe we should just remove this virtual env parens stuff and just say that all following commands assume that you're inside a `pipenv shell` environment? It will save us from having to add extra text to everything.

I also realized there's a bunch of commands below like `heroku` (should we add this to dev deps?) and `pre-commit` that rely on the environment.

**Review by akirabaruah — COMMENTED** [review]
Still LGTM 👍

File: dev-guide.md
Comment (cs-miller) [reply dev-guide.md:142]: yeah im mixed on this, it wasn't consistent to begin with. i will remove

File: dev-guide.md
Comment (cs-miller) [reply dev-guide.md:15]: here i don't think it actually matters? you just need it to install pipenv, and i think most methods of installing pipenv actually pull in python if you don't have it already. i just included it here for completeness

**Pull request merged**
